utf8_decode/src/lib.rs

```diff
@@ -1,76 +1,126 @@
+        metadata.add_tag("nlp");
+        metadata.add_tag("bytes");
+
+        let input = TensorMetadata::new("bytes");
+        input.set_description("The string as UTF-8 encoded bytes");
+        let hint =
+            supported_shapes(&[ElementType::U8], DimensionsParam::Fixed(&[0]));
+        input.add_hint(&hint);
+        metadata.add_input(&input);
+
+        let output = TensorMetadata::new("string");
+        output.set_description("The decoded text.");
+        let hint = supported_shapes(
+            &[ElementType::Utf8],
+            DimensionsParam::Fixed(&[1]),
+        );
+        output.add_hint(&hint);
+        metadata.add_output(&output);
+
+        register_node(&metadata);
+    }
 
-    fn transform(&mut self, input: Tensor<u8>) -> Self::Output {
-        let underlying_bytes: &[u8] = input.elements();
+    fn graph(node_id: String) -> Result<(), GraphError> {
+        let ctx = GraphContext::for_node(&node_id)
+            .ok_or(GraphError::MissingContext)?;
 
-        let mut useful_bytes = &underlying_bytes[..underlying_bytes.len()];
-        if let Some(index) = underlying_bytes.iter().position(|&x| x == 0) {
-            useful_bytes = &underlying_bytes[..index];
-        }
+        ctx.add_input_tensor(
+            "bytes",
+            ElementType::U8,
+            DimensionsParam::Fixed(&[0]),
+        );
 
-        let input_text = core::str::from_utf8(useful_bytes)
-            .expect("Input tensor should be valid UTF8");
+        ctx.add_output_tensor(
+            "string",
+            ElementType::U8,
+            DimensionsParam::Fixed(&[1]),
+        );
 
-        let output_text = vec![Cow::Owned(input_text.to_string())];
+        Ok(())
+    }
 
-        Tensor::new_vector(output_text)
+    fn kernel(node_id: String) -> Result<(), KernelError> {
+        let ctx = KernelContext::for_node(&node_id)
+            .ok_or(KernelError::MissingContext)?;
+
+        let TensorResult {
+            element_type,
+            dimensions,
+            buffer,
+        } = ctx.get_input_tensor("bytes").ok_or_else(|| {
+            KernelError::InvalidInput(InvalidInput {
+                name: "bytes".to_string(),
+                reason: BadInputReason::NotFound,
+            })
+        })?;
+
+        match element_type {
+            ElementType::U8 => {
+                buffer.view::<u8>(&dimensions).map_err(|e| {
+                    KernelError::InvalidInput(InvalidInput {
+                        name: "bytes".to_string(),
+                        reason: BadInputReason::InvalidValue(e.to_string()),
+                    })
+                })?;
+            },
+            other => {
+                return Err(KernelError::Other(format!(
+                "The Utf8 Decode proc-block doesn't support {:?} element type",
+                other,
+                )))
+            },
+        };
+
+        let output = transform(buffer.elements());
+
+        ctx.set_output_tensor(
+            "phrases",
```

Comment:
This was called `string` in the `graph()` function and in the metadata. 

```suggestion
            "string",
```